core/vm/ipgraph.go

```diff
@@ -10,10 +10,17 @@ import (
 	"github.com/ethereum/go-ethereum/log"
 )
 
+const (
+	ipGraphWriteGas        = 100
+	ipGraphReadGas         = 10
+	averageAncestorIpCount = 30
```

Comment:
Where do these constant come from?

Reply:
Good question! It currently based on the estimation of average ancestor count of each ip and further fine tune the gas cost is tracking an other ticket mentioned above.